Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submission-results/SubmissionResults.tsx

```diff
@@ -1,83 +1,38 @@
 import * as React from 'react';
 import { useCallback } from 'react';
-import Heading, { HeadingType } from '../../guidelines/headings/Heading';
-import Diff from '../../Diff';
 import { useSubmissionsDetails } from '../../../hooks/submissions/use-submissions-details';
 import { ITestRunDetailsType } from '../../../hooks/submissions/types';
-import styles from './SubmissionResults.module.scss';
+import TestRunDetails from '../test-run-details/TestRunDetails';
 
 const SubmissionResults = () => {
     const { currentSubmission } = useSubmissionsDetails();
 
-    const getTestRunHeading = (run: ITestRunDetailsType, runIndex: number) => `Test #${runIndex} (${run.resultType}) `;
+    const renderTestRunsDetails = useCallback((testRuns: ITestRunDetailsType[]) => testRuns.map((run, index) => (
+        <TestRunDetails
+          testRun={run}
+          testRunIndex={index + 1}
+        />
+    )), []);
 
-    const getIsCorrectAnswerResultType = (run: ITestRunDetailsType) => run.resultType === 'CorrectAnswer';
-
-    const getTestRunHeadingClassName = useCallback(
-        (run: ITestRunDetailsType) => (getIsCorrectAnswerResultType(run)
-            ? styles.correctTestRunHeading
-            : styles.wrongTestRunHeading),
-        [],
-    );
-
-    const renderTimeAndMemoryUsed = (run: ITestRunDetailsType) => (
-        <>
-            <p>
-                Time used:
-                {run.maxUsedTime}
-                s
-            </p>
-            <p>
-                Memory used:
-                {run.maxUsedMemory}
-                {' '}
-                MB
-            </p>
-        </>
+    const filterRuns = useCallback(
+        (trial: boolean) => currentSubmission!.testRuns.filter((tr) => tr.isTrialTest === trial),
```

Comment:
I would pass `testRuns` as a parameter. That way the exclamation mark will not be needed, as the null check for currentSubmission will be at the same function where the variable will be used.